File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int maxVol=0;
        int left = 0;
        int right = height.size()-1;

        while (left < right) {
            int vol = (right - left) * min(height[left], height[right]);
            maxVol = max(vol, maxVol);

            if (height[left] < height[right]) {
                while (left < right && height[left] >= height[left+1]){
                    left++;
                }
                left++;
            }
            else {
                while (left < right && height[right] >= height[right-1]) {
                    right--;
                }
                right--;
            }
        }
        return maxVol;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        max_area = 0
        n = len(height)
        left = 0
        right = n-1
        
        while (left < right):
            area = min(height[left], height[right]) * (right - left)
            if max_area < area:
                max_area = area
                
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        
        return max_area
        